// next/dist/build/turborepo-access-trace/result.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "TurborepoAccessTraceResult", {
    enumerable: true,
    get: function() {
        return TurborepoAccessTraceResult;
    }
});
class TurborepoAccessTraceResult {
    constructor(envVars = new Set([]), addresses = [], fsPaths = new Set([])){
        thisREMOVED_SECRETSVars = envVars;
        this.addresses = addresses;
        this.fsPaths = fsPaths;
    }
    /**
   * Merge another `TurborepoAccessTraceResult` into this one, mutating this `TurborepoAccessTraceResult`.
   */ merge(other) {
        otherREMOVED_SECRETSVars.forEach((envVar)=>thisREMOVED_SECRETSVars.add(envVar));
        other.fsPaths.forEach((path)=>this.fsPaths.add(path));
        this.addresses.push(...other.addresses);
        return this;
    }
    /**
   * Serialize this `TurborepoAccessTraceResult` into a serializable object. Used for passing
   * the `TurborepoAccessTraceResult` between workers where Sets are not serializable.
   */ serialize() {
        return {
            fs: Array.from(this.fsPaths).map(String),
            addresses: this.addresses,
            envVars: Array.from(thisREMOVED_SECRETSVars).map(String)
        };
    }
    /**
   * Squash this `TurborepoAccessTraceResult` into a public trace object that can be written to a file
   */ toPublicTrace() {
        return {
            network: this.addresses.length > 0,
            envVarKeys: Array.from(thisREMOVED_SECRETSVars).map(String),
            filePaths: Array.from(this.fsPaths).map(String)
        };
    }
    /**
   * Create an `TurborepoAccessTraceResult` from a serialized `SerializableTurborepoAccessTraceResult`
   */ static fromSerialized(serialized) {
        return new TurborepoAccessTraceResult(new Set(serializedREMOVED_SECRETSVars), serialized.addresses, new Set(serialized.fs));
    }
}
